cpp/src/svm/svc.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cublas_v2.h>
+#include "common/cumlHandle.hpp"
+#include "gram/kernelparams.h"
+
+namespace ML {
+namespace SVM {
+
+using namespace MLCommon;
+
+/**
+ * @brief C-Support Vector Classification
+ *
+ * This is an Scikit-Learn like wrapper around the stateless C++ functions.
+ *
+ * The classifier will be fitted using the SMO algorithm in dual space.
+ *
+ * The decision function takes the following form
+ * \f[
+ *    f(x) = sign\left( \sum_i=1^{N_{support} y_i \alpha_i K(x_i,x) + b \right),
+ * \f]
+ * where \f$x_i\f$ are the support vectors, and \f$ y_i \alpha_i \f$ are the dual
+ * coordinates.
+ *
+ * The penalty parameter C limits the values of the dual coefficients
+ * \f[ 0 <= \alpha <= C \f]
+ *
+ */
+template <typename math_t>
+class SVC {
```

Comment:
Need to provide fully stateless functions for Cython. The stateful class should only be calling down to the stateless functions and maintaining the intermediate state (eg params, cumlHandle)